framework/src/utils/RankFourTensor.C

```diff
@@ -12,6 +12,8 @@
 
 template class RankFourTensorTempl<Real>;
 template class RankFourTensorTempl<DualReal>;
+template class RankTwoTensorTempl<Complex>;
+template class RankFourTensorTempl<Complex>;
```

Comment:
The  `Complex` typedef is not known in MOOSE. This instantiation should be in Magpie for now.

Reply:
In [`ComplexTypes.C`](https://github.com/idaholab/magpie/blob/devel/src/utils/ComplexTypes.C) to be precise

Reply:
Can't `Complex` be available in MOOSE through libmesh: `libmesh_common.h`?

Reply:
Maybe, if libmesh was compiled with complex number support. But that's a bit besides the point. By design I'm trying to keep unnecessary explicit instantiations for complex types out of the main moose code base. Everything else related to complex types is in Magpie, so this belongs there, too. Maybe at some point in the future we'll have proper complex number support for the FEM parts of MOOSE and these instantiations will be useful in MOOSE, but I don't think thats the case quite yet.

Reply:
> if libmesh was compiled with complex number support.

I think that typedef is there all the time. I understood your point.